spec/src/main/asciidoc/core/packagingdeployment_lite.asciidoc

```diff
@@ -2,8 +2,116 @@
 
 == Packaging and deployment
 
-// TODO
-[NOTE]
-====
-TBD based on what approach we choose
-====
+When an application is started, the container must perform _bean discovery_, detect definition errors and deployment problems and raise events that allow build compatible extensions to integrate with the container initialization lifecycle.
+
+Bean discovery is the process of determining:
+
+* Bean archives within application, and any beans contained within them
+* Which alternatives and interceptors are _enabled_
+* The _ordering_ of enabled interceptors
+
+Additional beans may be registered programmatically via <<spi_lite>>.
+
+[[bean_archive_lite]]
+
+=== Bean archives
+
+Bean classes of enabled beans must be deployed in _bean archives_.
+
+A bean archive has a _bean discovery mode_ of either `annotated` or `none`.
+This is governed by presence of `beans.xml` file which can be either empty or it can declare the `bean-discovery-mode` attribute.
+Default value for this attribute is `annotated`.
+
+An archive which:
+
+* contains a `beans.xml` file with the `bean-discovery-mode` of `none`, or,
+* contains a portable or build compatible extension and no `beans.xml` file
+
+is not a bean archive.
+
+An _implicit bean archive_ is:
+
+* an archive which contains a `beans.xml` file that is empty, or,
+* an archive which contains a `beanx.xml` file that has `bean-discovery-mode` attribute set to `annotated`
+
+When determining which archives are bean archives, the container must consider:
+
+* Library jars
+* Directories in the JVM classpath
+
+Non Jakarta EE containers may or may not provide support for war, EJB jar or rar bean archives.
+
+The `beans.xml` file must be named:
+
+* `META-INF/beans.xml`.
+
+The container searches for beans in all bean archives in the application classpath.
```

Comment:
Classpath is again a runtime concept. Following my suggestion above, I'd just say:

```
The container searches for beans in all bean archives discovered as described above.
```